pachyderm/pachyderm
# *DO NOT MERGE* - redirect internal dex requests to external dex

This is a guide for how to set this up and iterate (synthesis with #10326):
1. Regenerate the Envoy config
`bazel run //etc/helm/pachyderm:update_envoy_config`
Configs get generated to the following directory (within the pachyderm repo):
`etc/helm/pachyderm/envoy.json`
2. Create the pachdev cluster
`bazel run //src/testing/pachdev create-cluster`
3. Install dex. Currently, I'm using vanilla dex with a mock connector.
Use the following command:
`helm install dex-test dex/dex --values dex.yaml --wait`
4. Build and push the console image (change the tag in the load-image command for every new build):
```
docker buildx create --name hab --driver docker-container  # only need to run once
docker buildx build --builder hab --build-arg DOCKER_TAG="local" -o type=docker -t pachyderm/haberdashery:local -f Dockerfile.console . 
bazel run //src/testing/pachdev load-image docker-daemon:pachyderm/haberdashery:local console:local00
```
5. Create a `~/.config/pachdev/pachyderm-values.json` file with the following:
```json
{
  "console": {
    "image": {
      "repository": "localhost:5001/console",
      "tag": "local00"
    }
  },
  "oidc": {
    "issuerURI": "http://pachyderm.local/dex"
  }
}
```
6. push pachdev
`bazel run //src/testing/pachdev push`

With this setup, you should be able to complete the auth login workflow with both pachctl and console.
